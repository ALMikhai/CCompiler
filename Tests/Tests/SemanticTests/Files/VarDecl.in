int a;
float b;
int *c;
const volatile float c1;
int * const * volatile b1;
