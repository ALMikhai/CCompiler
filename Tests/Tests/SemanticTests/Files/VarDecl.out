\-TRANSLATIONUNIT
  |-DECL
  | |-DECLSPEC
  | | |-INT
  | \-INITDECLARATORLIST
  |   \-INITDECLARATOR
  |     \-DECLARATOR
  |       \-a
  |-DECL
  | |-DECLSPEC
  | | |-FLOAT
  | \-INITDECLARATORLIST
  |   \-INITDECLARATOR
  |     \-DECLARATOR
  |       \-b
  |-DECL
  | |-DECLSPEC
  | | |-INT
  | \-INITDECLARATORLIST
  |   \-INITDECLARATOR
  |     \-DECLARATOR
  |       |-POINTER
  |       \-c
  |-DECL
  | |-DECLSPEC
  | | |-CONST
  | | \-DECLSPEC
  | |   |-VOLATILE
  | |   \-DECLSPEC
  | |     |-FLOAT
  | \-INITDECLARATORLIST
  |   \-INITDECLARATOR
  |     \-DECLARATOR
  |       \-c1
  \-DECL
    |-DECLSPEC
    | |-INT
    \-INITDECLARATORLIST
      \-INITDECLARATOR
        \-DECLARATOR
          |-POINTER
          | |-TYPEQUALIFIERLIST
          | | \-CONST
          | \-POINTER
          |   |-TYPEQUALIFIERLIST
          |   | \-VOLATILE
          \-b1

Structs { }
Symbols {
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: printf
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: scanf
	INT :: a
	FLOAT :: b
	POINTER to INT :: c
	const volatile FLOAT :: c1
	const POINTER to volatile POINTER to INT :: b1
}
